src/app/modules/products/server/procedure.ts

```
@@ -0,0 +1,61 @@
+          subcategories: (doc.subcategories?.docs ?? []).map((doc) => ({
+            ...(doc as Category),
+            subcategory: undefined,
+          })),
+        }));
+
+        const subcategoriesSlug = [];
+        const parentCategory = formattedData[0];
+
+        if (parentCategory) {
+          subcategoriesSlug.push(
+            ...(parentCategory.subcategories?.docs.map((subcategory) => subcategory.slug)
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)